File: Python/2108.py
maior = ''

while True:
    cc = ''
    pal = input().split()
    if '0' in pal[0]:
        break
    else:
        for i in pal:
            cc += f'{len(i)}-'
            if len(i) >= len(maior): maior = i
        print(cc[:-1])
print()
print(f'The biggest word: {maior}')

File: Java/2108.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String maior = "";
        while(sc.hasNextLine()){
            String cc = "";
            String[] pal = sc.nextLine().split(" ");
            if(pal[0].contains("0")){
                break;
            }
            for(int i = 0; i < pal.length; i++){
                if(i == pal.length-1)
                    System.out.println(pal[i].length());
                else
                    System.out.printf("%d-", pal[i].length());
                if(pal[i].length() >= maior.length())
                    maior = pal[i];
            }
        }
        System.out.println();
        System.out.println("The biggest word: " + maior);
        
    }
}
